improver/expected_value.py

```diff
@@ -72,24 +71,20 @@ def integrate_over_thresholds(cube: Cube) -> Cube:
         threshold_coord = find_threshold_coordinate(cube)
         threshold_coord_idx = cube.dim_coords.index(threshold_coord)
         thresholds = threshold_coord.points
+        # check the thresholds are in increasing order
+        if thresholds[0] > thresholds[-1]:
+            raise ValueError("threshold coordinate in decreasing order")
```

Comment:
This assumes the thresholds are either in increasing or decreasing order, but do you need to check for the case where they are not in _any_ order?